component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/jwt/JWTBearerGrantHandler.java

```diff
@@ -104,6 +107,38 @@ public void init() throws IdentityOAuth2Exception {
         }
     }
 
+    /**
+     * get resident Identity Provider
+     *
+     * @param tenantDomain tenant Domain
+     * @param jwtIssuer    issuer extracted from assertion
+     * @return resident Identity Provider
+     * @throws IdentityOAuth2Exception
+     */
+    private IdentityProvider getIdentityProvider(String tenantDomain, String jwtIssuer) throws IdentityOAuth2Exception {
+        String issuer = StringUtils.EMPTY;
+        IdentityProvider residentIdentityProvider = null;
+        try {
+            residentIdentityProvider = IdentityProviderManager.getInstance().getResidentIdP(tenantDomain);
+        } catch (IdentityProviderManagementException e) {
+            String errorMsg = String.format(ERROR_GET_RESIDENT_IDP, tenantDomain);
+            throw new IdentityOAuth2Exception(errorMsg, e);
+        }
+        FederatedAuthenticatorConfig[] fedAuthnConfigs = residentIdentityProvider.getFederatedAuthenticatorConfigs();
+        FederatedAuthenticatorConfig oauthAuthenticatorConfig =
+                IdentityApplicationManagementUtil.getFederatedAuthenticator(fedAuthnConfigs,
+                        IdentityApplicationConstants.Authenticator.OIDC.NAME);
+        if (oauthAuthenticatorConfig != null) {
+            issuer = IdentityApplicationManagementUtil.getProperty(oauthAuthenticatorConfig.getProperties(),
+                    OPENID_IDP_ENTITY_ID).getValue();
+        }
+        if (jwtIssuer.equals(issuer)) {
+            return residentIdentityProvider;
+        } else {
+            return null;
+        }
```

Comment:
Agree. 

A small concern I have is that this connector is intended to be used for IS 5.3.0 / APIM 2.1.0 which are C4 products. If we use Optional this connector will not work on Java7 right?